src/main/java/frc/robot/subsystems/conveyer/Conveyer.java

```diff
@@ -0,0 +1,130 @@
+   */
+  private SparkLimitSwitch beamBreakSensorOne,
+      beamBreakSensorTwo,
+      beamBreakSensorThree,
+      beamBreakSensorFour;
+
+  public enum NotePosition {
+    NO_NOTE,
+    RECEIVING_NOTE,
+    HOLDING_NOTE
+  }
+
+  public NotePosition notePosition = NotePosition.NO_NOTE;
+
+  public double secondMotorZeroPosition;
+
+  public Conveyer() {
+    SparkMaxUtils.initWithRetry(this::setUpConveyerSparks, ConveyerCal.SPARK_INIT_RETRY_ATTEMPTS);
+
+    beamBreakSensorOne = frontMotor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+    beamBreakSensorTwo = frontMotor.getReverseLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+    beamBreakSensorThree = backMotor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+    beamBreakSensorFour = backMotor.getReverseLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+
+    beamBreakSensorOne.enableLimitSwitch(false);
+    beamBreakSensorTwo.enableLimitSwitch(false);
+    beamBreakSensorThree.enableLimitSwitch(false);
+    beamBreakSensorFour.enableLimitSwitch(false);
+
+    secondMotorZeroPosition = backMotorEncoder.getPosition();
+  }
+
+  /** Does all the initialization for the sparks, return true on success */
+  private boolean setUpConveyerSparks() {
+    int errors = 0;
+    errors += SparkMaxUtils.check(frontMotor.restoreFactoryDefaults());
+    errors += SparkMaxUtils.check(backMotor.restoreFactoryDefaults());
+
+    errors += SparkMaxUtils.check(frontMotor.setIdleMode(IdleMode.kBrake));
+    errors += SparkMaxUtils.check(backMotor.setIdleMode(IdleMode.kBrake));
+
+    frontMotor.setInverted(ConveyerConstants.FRONT_MOTOR_INVERTED);
+    backMotor.setInverted(ConveyerConstants.BACK_MOTOR_INVERTED);
+
+    errors +=
+        SparkMaxUtils.check(
+            frontMotor.setSmartCurrentLimit(ConveyerCal.CONVEYER_CURRENT_LIMIT_AMPS));
+    errors +=
+        SparkMaxUtils.check(
+            backMotor.setSmartCurrentLimit(ConveyerCal.CONVEYER_CURRENT_LIMIT_AMPS));
+
+    return errors == 0;
+  }
+
+  public void prepareToShoot() {
+    frontMotor.set(ConveyerCal.PREPARE_TO_SHOOT_FRONT_SPEED);
+    backMotor.set(ConveyerCal.PREPARE_TO_SHOOT_BACK_SPEED);
+  }
+
+  public void scoreTrapOrAmp() {
+    frontMotor.set(ConveyerCal.SCORE_AMP_TRAP_FRONT_SPEED);
+    backMotor.set(ConveyerCal.SCORE_AMP_TRAP_BACK_SPEED);
+  }
+
+  public void hold() {
+    frontMotor.set(ConveyerCal.FRONT_HOLD_SPEED);
+    backMotor.set(ConveyerCal.BACK_HOLD_SPEED);
+  }
+
+  public void stop() {
+    frontMotor.set(0.0);
+    backMotor.set(0.0);
+  }
+
+  public void startReceivingNote() {
+    notePosition = NotePosition.RECEIVING_NOTE;
+    secondMotorZeroPosition = backMotorEncoder.getPosition();
+    hold();
+  }
+
+  /**
+   * Get the current position of the note. While we have the beam break sensors available, we are
+   * attempting to not use them, and rather just check the note position with the motors.
+   */
+  public NotePosition getNotePosition() {
```

Comment:
When does this get called should it be in periodic or would it be called from a command

Reply:
The logic is a little confusing depending on when this gets called

Reply:
This would be called in a command; I was thinking it would be like `WaitUntil(getNotePosition() == NotePosition.HOLDING_NOTE)`